packages/grafana-ui/src/components/Table/TableNG/Cells/PillCell.tsx

```diff
@@ -0,0 +1,98 @@
+import { classicColors } from '@grafana/data';
+
+import { PillCellProps } from '../types';
+
+function normalizePillValues(value: unknown): string[] {
+  if (Array.isArray(value)) {
+    return value.map(String).map(stripQuotes);
+  }
+  if (typeof value === 'string') {
+    // Try JSON parse
+    try {
+      const arr = JSON.parse(value);
+      if (Array.isArray(arr)) {
+        return arr.map(String).map(stripQuotes);
+      }
+    } catch (e) {
+      // Not JSON, continue
+    }
+    // Split by comma, but keep quoted substrings together
+    // e.g. 'foo,"Tim Levett",bar' => ['foo', 'Tim Levett', 'bar']
+    const regex = /(?:"([^"]*)")|(?:'([^']*)')|([^,]+)/g;
+    const pills: string[] = [];
+    let match;
+    while ((match = regex.exec(value)) !== null) {
+      const pill = match[1] || match[2] || match[3];
+      if (pill && pill.trim() !== '') {
+        pills.push(stripQuotes(pill.trim()));
+      }
+    }
+    if (pills.length > 0) {
+      return pills;
+    }
+    // Fallback: split by whitespace
+    if (value.match(/\s/)) {
+      return value.split(/\s+/).map(s => stripQuotes(s.trim())).filter(Boolean);
+    }
+    // Otherwise, treat as single pill if not empty
+    if (value.trim() !== '') {
+      return [stripQuotes(value.trim())];
+    }
+  }
+  return [];
+}
+
+// Helper to strip leading/trailing quotes
+function stripQuotes(str: string): string {
+  return str.replace(/^['"]+|['"]+$/g, '');
+}
+
+// Utility to determine if a color is "light" or "dark" for text contrast
+function isColorLight(hex: string): boolean {
```

Comment:
we have a function for this already: https://github.com/grafana/grafana/blob/4374fe9137f85484c334ed914cf1e6b8b10fe9c7/packages/grafana-data/src/themes/colorManipulator.ts#L214